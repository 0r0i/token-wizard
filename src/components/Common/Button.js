import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
