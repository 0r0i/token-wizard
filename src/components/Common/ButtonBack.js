import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
