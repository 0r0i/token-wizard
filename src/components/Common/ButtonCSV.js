import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
