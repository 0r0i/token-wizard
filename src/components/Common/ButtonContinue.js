import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
