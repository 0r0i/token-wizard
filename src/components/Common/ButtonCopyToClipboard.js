import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
