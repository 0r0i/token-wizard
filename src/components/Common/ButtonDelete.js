import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
