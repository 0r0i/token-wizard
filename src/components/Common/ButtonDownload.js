import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
