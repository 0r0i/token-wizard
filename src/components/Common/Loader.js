import React from 'react'

export const Loader = ({ show }) => {
  return show ? (
    <div className="sw-FullscreenBackdrop">
      <div className="sw-Loader">
        <div className="sw-Loader_AnimatedItem" />
        <div className="sw-Loader_AnimatedItem" />
        <div className="sw-Loader_AnimatedItem" />
        <div className="sw-Loader_AnimatedItem" />
        <div className="sw-Loader_AnimatedItem" />
        <div className="sw-Loader_AnimatedItem" />
      </div>
    </div>
  ) : null
}
export default Loader
