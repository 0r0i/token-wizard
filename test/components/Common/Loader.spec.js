import React from 'react'
import { Loader } from '../../../src/components/Common/Loader'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'

configure({ adapter: new Adapter() })

describe('Loader ', () => {
  it(`should render Loader component with class name 'loading container'`, () => {
    const wrapper = shallow(
      <Loader
        show={true}
      />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('[className="loading-container"]')).toBeDefined()
  })

  it(`should render Loader component with class name 'loading container'`, () => {
    const wrapper = shallow(
      <Loader
        show={false}
      />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('[className="loading-container notdisplayed"]')).toBeDefined()
  })
})
