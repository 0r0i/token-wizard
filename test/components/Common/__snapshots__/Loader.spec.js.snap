// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader  should render Loader component with class name 'loading container' 1`] = `
<div
  className="loading-container"
>
  <div
    className="loading"
  />
  <div
    className="loading-text-container"
  >
    <div
      className="loading-text"
    >
      Do not refresh the webpage
    </div>
  </div>
  <div
    className="loading-progress"
  >
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
  </div>
</div>
`;

exports[`Loader  should render Loader component with class name 'loading container' 2`] = `
<div
  className="loading-container notdisplayed"
>
  <div
    className="loading"
  />
  <div
    className="loading-text-container"
  >
    <div
      className="loading-text"
    >
      Do not refresh the webpage
    </div>
  </div>
  <div
    className="loading-progress"
  >
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
    <div
      className="loading-i"
    />
  </div>
</div>
`;
